Add extra Salt Bundle details (#1458)
uyuni-project/uyuni-docs: modules/specialized-guides/pages/salt/salt-ssh.adoc

As with all Salt calls, {productname} invokes `salt-ssh` via the `salt-api`.

Salt SSH relies on a roster to obtain details such as hostname, ports, and the SSH parameters of a client.
{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.

> The location of the temporary roster file is supplied to `salt-ssh` using the `--roster-file=` option.

Reviewer: ok, your explanation make sense (of course!).  In the long run, we should write "Salt SSH" when ever the subsystem is meant.  But this can come later, separately.  I think we now can merge your PR.  I unfortunately forgot again whether you have merge rights or not.  I can merge it for you.

> For registered clients `--roster=uyuni` is used instead.

Author: Actually the same as above. No `--` needed.

> For registered clients `--roster=uyuni` is used instead.

Reviewer: ok

[IMPORTANT]
====
It is not recommended to use [command]``salt-ssh`` as the call will be processed with current user permissions and in case of calling it with `root` user can cause permission issues for the files on processing next calls with `salt-api`.

> Please use [command]``mgr-salt-ssh`` instead. It is changing effective user to `salt` on each call. If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.

Reviewer: ```suggestion
Use [command]``mgr-salt-ssh`` instead because it is changing effective user to `salt` on each call.
If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.
```

As with all Salt calls, {productname} invokes `salt-ssh` via the `salt-api`.

Salt SSH relies on a roster to obtain details such as hostname, ports, and the SSH parameters of a client.
{productname} keeps these details in the database and makes them available to Salt by using `uyuni` roster module or by generating a temporary roster file on bootstrapping new clients with {webui} using Salt SSH call.

> The location of the temporary roster file is supplied to `salt-ssh` using the [option]``roster-file`` option.

Author: Actually `salt-ssh` is not a command in this case, in almost all the cases it's called in `salt-api` process.
So maybe the following is more correct (as it cold be in CLI as well as in salt-api event and the option has the same name for both cases):
```suggestion
The location of the temporary roster file is supplied to Salt SSH using the [option]``roster-file`` option.
```

For the registered clients [option]``roster`` set to `uyuni` is used instead to get the roster from the database with `uyuni` salt roster module.

[IMPORTANT]
====
It is not recommended to use [command]``salt-ssh`` as the call will be processed with current user permissions and in case of calling it with `root` user can cause permission issues for the files on processing next calls with `salt-api`.

> Use [command]``mgr-salt-ssh`` instead because it is changing effective user to `salt` on each call.

Reviewer 2: ```suggestion
Use [command]``mgr-salt-ssh``, which changes the effective user to [literal]``salt``and avoids file permission issues.
```

Author: ```suggestion
Use [command]``mgr-salt-ssh``, which changes the effective user to [literal]``salt``and avoids file permission issues. In addition it sets [option]``roster`` to [literal]``uyuni``, if neither [option]``roster`` nor [option]``roster-file`` specified, to allow using {productname} database as a source of roster data for the registered Salt SSH clients.
```

Reviewer 2: I thought that this "important" section should be kept short and direct. Everything you wrote is technically correct, I just wonder if it could distract from the message `use mgr-salt-ssh` due to its length.

Reviewer 3: I see @agraul point here. I would prefer something short and effective sentence, then we can go and add the technical detail in a differente sentence maybe.

> If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.

Author: It's not actually related to the previous one. `mgr-salt-ssh` can't be used by any other user as it can't switch the effective user to `salt` in this case. It's mostly about this permission requirement.

> If you want to use it with a user other than `root`, the user should have the permission to change effective user to `salt`.

Reviewer 2: It wasn't clear to me what "it" referred to. Then I would replace "it" with  

    [command]``mgr-salt-ssh``

Password authentication is used only when bootstrapping.
During the bootstrap step the key of the server is not authorized on the client and therefore a password must be used for a connection to be made.

> The password is used transiently in a temporary roster file used for bootstrapping and the roster file is removed on finishing processing the event.

Reviewer 2: IMO this change is not needed since "temporary roster file" implies that it's being removed afterwards.

Author: Some temporary files could be left untouched on the system and from the security point of view it's better to mention it explicitly.

Reviewer 2: You're right that "temporary" only means "not forever" and could imply a file is kept for 1 day / week / year etc. If you'd rather be explicit, that's fine for me :+1:

* `ssh_options`: other SSH options:
** `ProxyCommand` if the client connects through a proxy
* `timeout`: defaults to 180 seconds

> * `ssh_pre_flight`: The path to the pre flight shell script (default: /usr/share/susemanager/salt-ssh/preflight.sh)

Reviewer: ```suggestion
* `ssh_pre_flight`: The path to the pre flight shell script (default: [path]``/usr/share/susemanager/salt-ssh/preflight.sh``)
```
